src/main/java/virushade/tasks/TaskList.java

```diff
@@ -0,0 +1,210 @@
+
+    /**
+     * This variable keeps track of the size of the TaskList.
+     */
+    private static int listCount = 0;
+
+    /**
+     * Constructor for the TaskList class.
+     * @param storage The storage for classList to load data from.
+     * @throws VirushadeException Thrown when there is trouble parsing the data file in storage.
+     */
+    public TaskList(Storage storage) throws VirushadeException {
+        TASK_LIST_STORAGE = storage;
+        storage.load(TASKS);
+        listCount = TASKS.size();
+    }
+
+    /**
+     * A function that writes an input string to TASK_LIST_STORAGE.
+     *
+     * @param text The input string to write into the file.
+     */
+    private static void updateFile(String text) throws VirushadeException {
+        TASK_LIST_STORAGE.update(text);
+    }
+
+    /**
+     * Adds a Task to taskList.
+     * @param addedTaskDescription Name of the added Task.
+     * @param taskType The type of task to be added.
+     * @throws VirushadeException Thrown when there is trouble adding the task.
+     */
+    public static void add(String addedTaskDescription, String taskType) throws VirushadeException {
+        if (listCount < 100) {
+            Task addedTask;
+            String[] strings = StringManipulator.slashPartition(addedTaskDescription);
+
+            // Checking if the Task Name is empty (Or filled with Spacebars):
+            if (strings[0].replace(" ", "").equals("")) {
+                throw new VirushadeException("OOPS!!! Please enter the task description!!!");
+            }
+
+            switch (taskType) {
+            case "TODO":
+                addedTask = new ToDo(addedTaskDescription, false);
+                break;
+
+            case "DEADLINE":
+                if (strings[1].startsWith("by ")) {
+                    addedTask = new Deadline(strings[0], strings[1].substring(3), false);
+                } else {
+                    System.out.println("Please include a deadline after your task name. " +
+                            "(e.g. /by Sunday)");
+                    return;
+                }
+                break;
+
+            case "EVENT":
+                if (strings[1].startsWith("at ")) {
+                    addedTask = new Event(strings[0], strings[1].substring(3), false);
+                } else {
+                    System.out.println("Please include a time after your task name. " +
+                            "(e.g. /at 12 noon)");
+                    return;
+                }
+                break;
+
+            // The add function would not reach this line at all.
+            default:
+                addedTask = new Task(addedTaskDescription, false);
+            }
+
+            TASKS.add(addedTask);
+            listCount++;
+            System.out.println("Added: " + addedTask.getTaskDescription());
+            System.out.printf("Now you have %d tasks in the list.\n", listCount);
+            updateFile(generateList());
```

Comment:
Good idea to update the file everytime a task is added. That way your tasks will still be saved for recovery in case the application crashs.